server/constants/routes.go

```diff
@@ -0,0 +1,42 @@
+package constants
+
+const (
+	// Plugin API Routes
+	APIPrefix                               = "/api/v1"
+	WildRoute                               = "{anything:.*}"
+	PathOAuthConnect                        = "/oauth/connect"
+	PathOAuthCallback                       = "/oauth/complete"
+	PathLinkedProjects                      = "/project/link"
+	PathGetAllLinkedProjects                = "/project/link"
+	PathUnlinkProject                       = "/project/unlink"
+	PathUser                                = "/user"
+	PathCreateTasks                         = "/tasks"
+	PathLinkProject                         = "/link"
+	PathSubscriptions                       = "/subscriptions"
+	PathGetSubscriptions                    = "/subscriptions/{team_id:[A-Za-z0-9]+}"
+	PathSubscriptionNotifications           = "/notification"
+	PathGetUserChannelsForTeam              = "/channels/{team_id:[A-Za-z0-9]+}"
```

Comment:
actually routes `PathSubscriptions, PathGetSubscriptions` are handling the requests to store/delete and get subscriptions from KV store

`PathSubscriptionNotifications` is a plugin API that listens for notifications sent by the Azure DevOps

For `PathGetUserChannelsForTeam` we could have directly it called from the web app but we implemented this as a plugin API because we are checking and returning only public and private channels and not DM or group

Reply:
@avas27JTG I recommend using the server's existing endpoint from the frontend, and filter out the private channels and DMs on the frontend side

Reply:
@mickmister 
We are making use of RTK Query to make API calls which offer `createApi`([more details here](https://redux-toolkit.js.org/rtk-query/api/createApi)) function to hit endpoints of a base URL, we already have one [here](https://github.com/mattermost/mattermost-plugin-azure-devops/blob/6d00878f77da568f7c051b43d15faced3b3f12fd/webapp/src/services/index.ts#L11)

Now if we try calling server's endpoint directly from the webapp then this will need to create a new `createApi` function with the server base URL as its's endpoint will be different from the plugin also, it's advised to have only one `createApi` function as a best practice

also, filtering out the `groups and DMs` from the channel's list on the webapp would be slower than on the server

Please let us know if we still need to make this change

Reply:
> also, filtering out the groups and DMs from the channel's list on the webapp would be slower than on the server

@avas27JTG I think it's negligible in this case. A user is not going to have a massive amount of channel memberships. I just think it's wrong to duplicate an endpoint that already exists on the server.

> Now if we try calling server's endpoint directly from the webapp then this will need to create a new createApi function with the server base URL as its's endpoint will be different from the plugin also, it's advised to have only one createApi function as a best practice

It seems like a slippery slope to me that we can't use an existing endpoint because of complications with a library we're using. It's not a requirement to change this for this PR, but I highly advise against developing this way.

Reply:
Sure @mickmister
also, there aren't any complications with the library(RTK Query) for using multiple endpoints as we can change it easily to support that, it was just because of maintaining two endpoints/baseURL for a single web app, but I think in the case of Mattermost it makes sense to avoid the plugin's API where server's API is available, will change make this change

Reply:
@mickmister this has been fixed here https://github.com/mattermost/mattermost-plugin-azure-devops/pull/141/commits/49dad5a1ee98d035a496f94c9396d9d15fa74470